patches/tModLoader/Terraria/ModLoader/Core/ModOrganizer.cs

```diff
@@ -508,6 +510,8 @@ private enum SearchFolders { }
 		private static readonly Regex PublishFolderMetadata = new Regex(@"[/|\\]([0-9]{4}[.][0-9]{1,2})[/|\\]");
 
 		internal static string GetActiveTmodInRepo(string repo) {
+			// TODO: This logic is incorrect. I think sometimes we want to know every mod, and othertimes we want to know every mod with a version we can load.
+			// The tmods.Length == 1 check and the logic below contradict each other, sometimes returning null when no loadable mods and othertimes returning a mod path.
 			Version tmodVersion = new Version(BuildInfo.tMLVersion.Major, BuildInfo.tMLVersion.Minor);
```

Comment:
This logic should probably be rewritten and refactored with the CleanupPublishFolder logic.

Specs on this function:
- tml version 0.12 is a direct file path to .tmod (no folder). Is a legacy case
- for purposes of mod loading, we want the latest .tmod for the tml version running
- if there is no .tmod for this version or earlier, we return null (ie no mod exists that can be loaded). Should be logged